book/core_programming.md

```diff
@@ -2,153 +2,310 @@
 
 The principles outlined in this chapter represent good practices for general programming and software development.
 
-Before reading this chapter, you would benefit from having an understanding of core programming concepts.
-This includes assigning and using variables, and using functions and classes.
+However, they are tailored to a more analytical workflow and would benefit analysts with understanding of core programming concepts such as variables, functions and classes. You can find more information and helpful resources in the [resources](resources) section.
 
+## Motivation
 
-```{admonition} Pre-requisites
-:class: admonition-learning
+```{epigraph}
+Code is read more often than it is written.
 
-You should look to these sources for introductory learning, before reading this guidance:
-* Introduction to [Python](https://learninghub.ons.gov.uk/enrol/index.php?id=536) and [R](https://learninghub.ons.gov.uk/enrol/index.php?id=538) (government analysts only)
-* Aggregation, functions and control flow in [Python](https://learninghub.ons.gov.uk/enrol/index.php?id=525) and [R](https://learninghub.ons.gov.uk/enrol/index.php?id=527) (government analysts only)
-* Official Python [Getting Started Guide](https://www.python.org/about/gettingstarted/)
-* [Learn Python](https://www.learnpython.org/) supported by DataCamp
-* [R Basics](https://www.udemy.com/course/r-basics/?LSNPUBID=JVFxdTr9V80&ranEAID=JVFxdTr9V80&ranMID=39197&ranSiteID=JVFxdTr9V80-mzkL.lpta.ugiFro.8Oc_Q&utm_medium=udemyads&utm_source=aff-campaign) on Udemy
-* [edx](https://www.edx.org/) and [freeCodeCamp](https://www.freecodecamp.org/) hosts many free online coding courses
-* [DataCamp](https://www.datacamp.com/), [Codecademy](https://www.codecademy.com/) and [Coursera](https://www.coursera.org/) host a range of introductory courses, but more advanced courses are often not free.
+-- Guido van Rossum (creator of Python)
+```
 
-Please note that non-government analysis learning may not follow good practices. However, exposure to a range of applied examples will still benefit your learning.
-You should compare and contrast your learning to the good practices outlined in this section.
+When writing non-trivial code it is wise to assume that at some point someone else will need to understand, use and adapt your code. It might be yourself in six months time - after you've forgotten how the code works, or its design intent. Therefore every time you write such code, it is incredibly important to empathise with these potential users and produce code that is tidy, understandable and does not add unnecessary complexity.
```

Comment:
I think we can remove "incredibly". "non-trivial" and "design intent" can also be removed. We should include something about the fact that you can't be expected to maintain EVERY piece of code you write too.

Reply:
I removed the words you mentioned, but I think the message here is - if you're writing code that someone else will need, you need to empathise with them. I've tweaked the wording, so will close this and you can point it out in the next review if you think its ok.